signature-specification.md

```diff
@@ -0,0 +1,519 @@
+        }
+    ]
+}
+```
+
+If a JWS contains only one signature as above, the JWS can be flattened as
+
+```json
+{
+    "payload": "<Base64Url(Payload)>",
+    "protected": "<Base64Url(ProtectedHeader)>",
+    "header": {
+        "timestamp": "<Base64(TimeStampToken)>",
+        "kid": "906ade40b96cff95e5b60f7e96f2cda7979c8ad5",
+        "x5c": [
+            "<Base64(DER(leafCert))>",
+            "<Base64(DER(intermediateCACert))>",
+            "<Base64(DER(rootCert))>"
+        ]
+    },
+    "signature": "Base64Url( Sign( ASCII( <Base64Url(ProtectedHeader)>.<Base64Url(Payload)> )))"
+}
+```
+
+- `protected`: Base64Url encoded JSON object that contains the header parameters that are integrity protected by the JWS Signature digital signature.
+- `header`: JOSE Header containing the parameters describing the cryptographic operations and parameters employed. header is not integrity protected by signature. To start with we will only support reserved set of headers.
+    - `timestamp`: Base64 encoded [timestamp token](https://datatracker.ietf.org/doc/html/rfc3161#section-2.4.2) generated by TSA. Use of this is OPTIONAL.
+    - `kid`: Hint indicating which key was used to generate the signature. Use of this is OPTIONAL.
+    - `x5c`: Contains the X.509 public key certificate or certificate chain corresponding to the key used to generate the signature. Use of this is OPTIONAL. If signature was generated by X.509 certificate signature envelop MUST contain `x5c`.
+- The `header` node MUST contain either `kid` or `x5c` but not both.
+- In case of `x5c`, the leaf certificate's public key algorithm (with some additional conventions) will be used for signature generation and this algorithm must match with `alg` header value. The verifier will make sure that the value of `alg` header is same as that of leaf certificate's signing algorithm.
+    - Additional convention for algorithm: If the X.509 certificate does not indicate the hash algorithm (as is the case of ECDSA), then the implementation will use hard-code a mapping similar to that of [JWS](https://datatracker.ietf.org/doc/html/rfc7518#section-3.1).
+- Implementation should only support predefined set of asymmetric signing algorithms, which means value of `alg` will always be from a predefined set of values.
+
+</details>
+
+
+### Dead Simple Signature Envelope ([DSSE](https://github.com/secure-systems-lab/dsse))
+
+Pros:
+
+- Simple, lesser probability of user shooting themselves in the foot.
+- Signing libraries available in Go: [in-toto-golang](https://github.com/in-toto/in-toto-golang/commit/5b9e472f1a6e04f14d62266dd8e24749c2e9900c?branch=5b9e472f1a6e04f14d62266dd8e24749c2e9900c&diff=split).
+
+Cons:
+
+- No published RFC (or draft) and not commonly used. Only known usage is [in-toto](https://in-toto.io) and that [implementation also diverges from specification](https://github.com/secure-systems-lab/dsse/issues/40#issuecomment-882771124).

```

Comment:
TUF and cosign are also DSSE users (or at least considering), and each of those are CNCF or LF projects that would benefit from interoperability with Notary. I would consider that list of projects to be a strong Pro. It also means any organization that gets the approval to use one of these other projects will have an easier time getting other DSSE projects approved for use by their review boards.